calls: 0.1.5 -> 0.2.0 with init of callaudiod and feedbackd

Pacman99 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
To update calls and add its new dependencies: callaudiod and feedbackd. 

The feedbackd package is from @masipcat's #94820, I just bumped the version. I can remove it if that PR gets merged first, but its necessary for this version of calls to build right.

Also callaudiod and feedbackd should probably be setup as modules at some point, possibly as a part of #88767. I might try my hand at it sometime.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Commit:
callaudiod: init at 0.0.4

Co-authored-by: davidak <davidak@users.noreply.github.com>

Commit:
feedbackd-unstable: init at 2020-01-25

SuperSandro2000 commented on pkgs/applications/audio/callaudiod/default.nix:
```diff
@@ -0,0 +1,42 @@
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgconfig
```
```suggestion
    pkg-config
```

SuperSandro2000 commented on pkgs/applications/audio/callaudiod/default.nix:
```diff
@@ -0,0 +1,42 @@
+, fetchFromGitLab
+, meson
+, ninja
+, pkgconfig
```
```suggestion
, pkg-config
```

SuperSandro2000 commented on pkgs/applications/misc/feedbackd/default.nix:
```diff
@@ -0,0 +1,51 @@
+    description = "A daemon to provide haptic (and later more) feedback on events";
+    homepage = "https://source.puri.sm/Librem5/feedbackd";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ masipcat pacman99 ];
```
Please add yourself to maintainers.nix.

Pacman99 replied:
I'm in the maintainer-list.nix, https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix#L6734

SuperSandro2000 replied:
Not sure why ofborg failed with this error. Maybe you where not at eval time.

Pacman99 replied:
Oh I just realized masipcat isn't yet on the list, I removed masipcat for now.

SuperSandro2000 commented:
Result of `nixpkgs-review pr 107681` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>callaudiod</li>
    <li>calls</li>
    <li>feedbackd</li>
  </ul>
</details>

davidak commented on pkgs/applications/misc/feedbackd/default.nix:
```diff
@@ -0,0 +1,51 @@
+
+stdenv.mkDerivation rec {
+  pname = "feedbackd";
+  version = "0.0.0+git20201114";
```
```suggestion
  version = "2020-11-14";
```

as written here https://nixos.org/manual/nixpkgs/stable/#sec-package-naming

davidak replied:
i see that is their official release name. wtf. nevermind

Commit:
calls: 0.1.5 -> 0.2.0

Pacman99 commented:
Updated calls to 0.2.0 and fixed name and version for feedbackd to match the naming rules. Callaudiod cannot be updated, because calls depends on 0.4.0, this should change in the next calls version.

davidak commented:
Result of `nixpkgs-review pr 107681` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>callaudiod</li>
    <li>calls</li>
    <li>feedbackd</li>
  </ul>
</details>

davidak commented on pkgs/applications/audio/callaudiod/default.nix:
```diff
@@ -0,0 +1,42 @@
+  ];
+
+  meta = with stdenv.lib; {
+    description = "a daemon for dealing with audio routing during phone calls";
```
```suggestion
    description = "Daemon for dealing with audio routing during phone calls";
```

meta.description should:

    Be capitalized.

that's what fedora uses and i think it makes sense to have the same description across distros, like the package names should also be the same

https://repology.org/project/callaudiod/packages

davidak reviewed (changes requested):
one minor nitpick

davidak commented:
Package builds and runs. Cant test the actual functionality.

![Screenshot from 2021-01-21 22-18-07](https://user-images.githubusercontent.com/91113/105414880-6194d580-5c38-11eb-8050-f9da2773132d.png)

One thing i noticed is that when i close it, the process still runs until i kill it with CTRL+C.

```
[nix-shell:~/.cache/nixpkgs-review/pr-107681]$ calls

(calls:27003): dbind-WARNING **: 22:18:50.186: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-GXwHDL6PHu: Connection refused

(sm.puri.Calls:27003): Gtk-CRITICAL **: 22:18:50.276: gtk_application_uninhibit: assertion 'GTK_IS_APPLICATION (application)' failed

^C
```

davidak reviewed (approved):
LGTM

Pacman99 commented:
Yeah the description thing makes sense to me, just curious does nixpkgs have any specific guidelines for descriptions? I usually just copy directly from nixpkgs. 

>One thing i noticed is that when i close it, the process still runs until i kill it with CTRL+C.

I think thats intended, it is supposed to be also be a daemon, so it doesn't kill on application close. Its so it can receive phone calls.

davidak commented:
> I usually just copy directly from upstream.

That makes sense most of the time. We have a few simple rules.

https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

>Its so it can receive phone calls.

i see. good

SuperSandro2000 commented on pkgs/applications/audio/callaudiod/default.nix:
```diff
@@ -0,0 +1,42 @@
+    glib
+  ];
+
+  meta = with stdenv.lib; {
```
```suggestion
  meta = with lib; {
```
This got recently deprecated and shouldn't be used for new packages.

SuperSandro2000 commented on pkgs/applications/misc/feedbackd/default.nix:
```diff
@@ -0,0 +1,51 @@
+    json-glib
+  ];
+
+  meta = with stdenv.lib; {
```
```suggestion
  meta = with lib; {
```

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107681` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>callaudiod</li>
<li>calls</li>
<li>feedbackd</li>
</ul>
</details>

Pacman99 commented:
Updated feedbackd to 2021-01-25

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107681` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>callaudiod</li>
<li>calls</li>
<li>feedbackd</li>
</ul>
</details>

Pacman99 commented:
Thanks for merging!
